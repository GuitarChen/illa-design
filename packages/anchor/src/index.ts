export * from "./anchor"
export * from "./interface"
