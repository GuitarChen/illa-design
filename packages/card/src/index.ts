export * from "./card"
export * from "./interface"
export * from "./card-grid"
export * from "./meta"
