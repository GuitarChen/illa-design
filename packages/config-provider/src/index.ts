export * from "./config-provider"
export * from "./interface"
export * from "./config-provider-context"
export * from "./config"

// locale
export * from "./locale/en-US"
export * from "./locale/zh-CN"
export * from "./locale/def"
