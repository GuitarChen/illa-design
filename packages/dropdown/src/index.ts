export * from "./dropdown"
export * from "./interface"
