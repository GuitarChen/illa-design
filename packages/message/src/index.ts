export * from "./style"
export * from "./message"
export * from "./interface"
