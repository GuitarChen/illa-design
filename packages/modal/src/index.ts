export * from "./modal"
export * from "./interface"
export * from "./style"
export * from "./portal"
export * from "./config"
export * from "./confirm"
export * from "./useModal/hookModal"
export * from "./useModal/contextHolder"
export * from "./useModal/index"
