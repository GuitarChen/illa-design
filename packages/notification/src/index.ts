export * from "./notice"
export * from "./interface"
export * from "./notification"
export * from "./style"
