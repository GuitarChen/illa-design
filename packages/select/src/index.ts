export * from "./select"
export * from "./select-view"
export * from "./option"
export * from "./interface"
export * from "./select-view"
export * from "./utils"
