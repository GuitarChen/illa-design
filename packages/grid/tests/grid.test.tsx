import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"

import { Col, Row } from "../src"

test("Grid renders correctly.", () => {
  render(
    <Row data-testid="test-row">
      <Col data-testid="test-col" />
      <Col />
    </Row>,
  )
  expect(screen.getByTestId("test-row")).toBeInTheDocument()
  expect(screen.getByTestId("test-col")).toBeInTheDocument()
})

test("Grid renders span.", () => {
  render(
    <Row
      data-testid="test-row"
      align="start"
      justify="end"
      horizontalGap="30px"
      verticalGap="30px"
    >
      <Col span={13} />
      <Col span={15} />
    </Row>,
  )
  expect(screen.getByTestId("test-row")).toMatchSnapshot()
})

test("Grid renders col correctly.", () => {
  render(
    <Row data-testid="test-row">
      <Col span={12} offset={3} order={2} push={3} pull={3} />
      <Col span={12} offset={3} order={1} push={3} pull={3} />
    </Row>,
  )
  expect(screen.getByTestId("test-row")).toMatchSnapshot()
})

test("Grid renders with gap.", () => {
  render(
    <Row
      data-testid="test-row"
      horizontalGap={{
        xs: "12px",
        sm: "12px",
        md: "12px",
        lg: "12px",
        xl: "12px",
        xxl: "12px",
      }}
      verticalGap={{
        xs: "12px",
        sm: "12px",
        md: "12px",
        lg: "12px",
        xl: "12px",
        xxl: "12px",
      }}
    >
      <Col span={20} />
      <Col span={15} />
    </Row>,
  )
  expect(screen.getByTestId("test-row")).toMatchSnapshot()
})

test("Grid renders reactive correctly.", () => {
  render(
    <Row data-testid="test-row">
      <Col
        xs={{
          span: 12,
          offset: 3,
          order: 2,
          push: 3,
          pull: 3,
        }}
        sm={{
          offset: 3,
          order: 2,
          push: 3,
          pull: 3,
        }}
        md={{
          span: 12,
          offset: 3,
          order: 2,
          push: 3,
          pull: 3,
        }}
        lg={{
          span: 12,
          offset: 3,
          order: 2,
          push: 3,
          pull: 3,
        }}
        xl={{
          span: 12,
          offset: 3,
          order: 2,
          push: 3,
          pull: 3,
        }}
        xxl={{
          span: 12,
          offset: 3,
          order: 2,
          push: 3,
          pull: 3,
        }}
      />
      <Col xs={12} sm={12} md={12} lg={12} xl={12} xxl={12} />
    </Row>,
  )
  expect(screen.getByTestId("test-row")).toMatchSnapshot()
})
