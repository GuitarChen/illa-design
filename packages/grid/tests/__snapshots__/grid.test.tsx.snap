// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders col correctly. 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-1 {
  box-sizing: border-box;
  position: relative;
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  margin-left: calc(3 * calc(100% / 24));
  width: calc(12 * calc(100% / 24));
}

.emotion-2 {
  box-sizing: border-box;
  position: relative;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  margin-left: calc(3 * calc(100% / 24));
  width: calc(12 * calc(100% / 24));
}

<div
  class="emotion-0"
  data-testid="test-row"
>
  <div
    class="emotion-1"
  />
  <div
    class="emotion-2"
  />
</div>
`;

exports[`Grid renders reactive correctly. 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-1 {
  position: relative;
  box-sizing: border-box;
  width: calc(24 * calc(100% / 24));
}

@media (min-width: 0px) {
  .emotion-1 {
    left: calc(3 * calc(100% / 24));
    right: calc(3 * calc(100% / 24));
    margin-left: calc(3 * calc(100% / 24));
    width: calc(12 * calc(100% / 24));
  }
}

@media (min-width: 576px) {
  .emotion-1 {
    left: calc(3 * calc(100% / 24));
    right: calc(3 * calc(100% / 24));
    margin-left: calc(3 * calc(100% / 24));
  }
}

@media (min-width: 768px) {
  .emotion-1 {
    left: calc(3 * calc(100% / 24));
    right: calc(3 * calc(100% / 24));
    margin-left: calc(3 * calc(100% / 24));
    width: calc(12 * calc(100% / 24));
  }
}

@media (min-width: 992px) {
  .emotion-1 {
    left: calc(3 * calc(100% / 24));
    right: calc(3 * calc(100% / 24));
    margin-left: calc(3 * calc(100% / 24));
    width: calc(12 * calc(100% / 24));
  }
}

@media (min-width: 1200px) {
  .emotion-1 {
    left: calc(3 * calc(100% / 24));
    right: calc(3 * calc(100% / 24));
    margin-left: calc(3 * calc(100% / 24));
    width: calc(12 * calc(100% / 24));
  }
}

@media (min-width: 1600px) {
  .emotion-1 {
    left: calc(3 * calc(100% / 24));
    right: calc(3 * calc(100% / 24));
    margin-left: calc(3 * calc(100% / 24));
    width: calc(12 * calc(100% / 24));
  }
}

.emotion-2 {
  position: relative;
  box-sizing: border-box;
  width: calc(24 * calc(100% / 24));
}

@media (min-width: 0px) {
  .emotion-2 {
    width: calc(12 * calc(100% / 24));
  }
}

@media (min-width: 576px) {
  .emotion-2 {
    width: calc(12 * calc(100% / 24));
  }
}

@media (min-width: 768px) {
  .emotion-2 {
    width: calc(12 * calc(100% / 24));
  }
}

@media (min-width: 992px) {
  .emotion-2 {
    width: calc(12 * calc(100% / 24));
  }
}

@media (min-width: 1200px) {
  .emotion-2 {
    width: calc(12 * calc(100% / 24));
  }
}

@media (min-width: 1600px) {
  .emotion-2 {
    width: calc(12 * calc(100% / 24));
  }
}

<div
  class="emotion-0"
  data-testid="test-row"
>
  <div
    class="emotion-1"
  />
  <div
    class="emotion-2"
  />
</div>
`;

exports[`Grid renders span. 1`] = `
.emotion-0 {
  -webkit-column-gap: 30px;
  column-gap: 30px;
  row-gap: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: end;
  justify-content: end;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.emotion-1 {
  position: relative;
  box-sizing: border-box;
  width: calc(13 * calc((100% - 30px * (2 - 1)) / 24));
}

.emotion-2 {
  position: relative;
  box-sizing: border-box;
  width: calc(15 * calc((100% - 30px * (2 - 1)) / 24));
}

<div
  class="emotion-0"
  data-testid="test-row"
>
  <div
    class="emotion-1"
  />
  <div
    class="emotion-2"
  />
</div>
`;

exports[`Grid renders with gap. 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

@media (min-width: 0px) {
  .emotion-0 {
    -webkit-column-gap: 12px;
    column-gap: 12px;
  }
}

@media (min-width: 576px) {
  .emotion-0 {
    -webkit-column-gap: 12px;
    column-gap: 12px;
  }
}

@media (min-width: 768px) {
  .emotion-0 {
    -webkit-column-gap: 12px;
    column-gap: 12px;
  }
}

@media (min-width: 992px) {
  .emotion-0 {
    -webkit-column-gap: 12px;
    column-gap: 12px;
  }
}

@media (min-width: 1200px) {
  .emotion-0 {
    -webkit-column-gap: 12px;
    column-gap: 12px;
  }
}

@media (min-width: 1600px) {
  .emotion-0 {
    -webkit-column-gap: 12px;
    column-gap: 12px;
  }
}

@media (min-width: 0px) {
  .emotion-0 {
    row-gap: 12px;
  }
}

@media (min-width: 576px) {
  .emotion-0 {
    row-gap: 12px;
  }
}

@media (min-width: 768px) {
  .emotion-0 {
    row-gap: 12px;
  }
}

@media (min-width: 992px) {
  .emotion-0 {
    row-gap: 12px;
  }
}

@media (min-width: 1200px) {
  .emotion-0 {
    row-gap: 12px;
  }
}

@media (min-width: 1600px) {
  .emotion-0 {
    row-gap: 12px;
  }
}

.emotion-1 {
  position: relative;
  box-sizing: border-box;
  width: calc(20 * calc(100% / 24));
}

.emotion-2 {
  position: relative;
  box-sizing: border-box;
  width: calc(15 * calc(100% / 24));
}

<div
  class="emotion-0"
  data-testid="test-row"
>
  <div
    class="emotion-1"
  />
  <div
    class="emotion-2"
  />
</div>
`;
